src/main/java/bio/terra/cli/service/utils/HttpUtils.java

```diff
@@ -122,52 +122,120 @@ public static HttpResponse sendHttpRequest(
-   * retries and {@link #DEFAULT_MILLISECONDS_SLEEP_FOR_RETRY} for the time to sleep between
-   * retries.
+   * Helper method to call a function with retries. Uses {@link #DEFAULT_MAXIMUM_RETRIES} for
+   * maximum number of retries and {@link #DEFAULT_DURATION_SLEEP_FOR_RETRY} for the time to sleep
+   * between retries.
    *
    * @param makeRequest function to perform the request
    * @param isRetryable function to test whether the exception is retryable or not
-   * @param <T> type of the Http response (i.e. return type of the makeRequest function)
-   * @return the Http response
-   * @throws Exception if makeRequest throws an exception that is not retryable, or if the maximum
-   *     number of retries was exhausted
+   * @param <T> type of the response object (i.e. return type of the makeRequest function)
+   * @return the response object
+   * @throws E if makeRequest throws an exception that is not retryable
+   * @throws SystemException if the maximum number of retries is exhausted, and the last attempt
+   *     threw a retryable exception
    */
   public static <T, E extends Exception> T callWithRetries(
       SupplierWithCheckedException<T, E> makeRequest, Predicate<Exception> isRetryable)
       throws E, InterruptedException {
     return callWithRetries(
-        DEFAULT_MAXIMUM_RETRIES, DEFAULT_MILLISECONDS_SLEEP_FOR_RETRY, makeRequest, isRetryable);
+        DEFAULT_MAXIMUM_RETRIES, DEFAULT_DURATION_SLEEP_FOR_RETRY, makeRequest, isRetryable);
   }
 
   /**
-   * Helper method to do retries.
+   * Helper method to call a function with retries.
    *
-   * @param maxRetries maximum number of times to retry
+   * @param maxCalls maximum number of times to retry
    * @param sleepDuration time to sleep between tries
    * @param makeRequest function to perform the request
    * @param isRetryable function to test whether the exception is retryable or not
-   * @param <T> type of the Http response (i.e. return type of the makeRequest function)
-   * @return the Http response
-   * @throws Exception if makeRequest throws an exception that is not retryable, or if the maximum
-   *     number of retries was exhausted
+   * @param <T> type of the response object (i.e. return type of the makeRequest function)
+   * @return the response object
+   * @throws E if makeRequest throws an exception that is not retryable
+   * @throws SystemException if the maximum number of retries is exhausted, and the last attempt
+   *     threw a retryable exception
    */
   public static <T, E extends Exception> T callWithRetries(
-      int maxRetries,
+      int maxCalls,
       Duration sleepDuration,
       SupplierWithCheckedException<T, E> makeRequest,
       Predicate<Exception> isRetryable)
       throws E, InterruptedException {
+    // isDone always return true
+    return pollWithRetries(maxCalls, sleepDuration, makeRequest, (result) -> true, isRetryable);
+  }
+
```

Comment:
This is just a nit, and feel free to ignore if this is my C++ bias showing, but I was surprised that the two parameters that get "defaulted" by overload (`DEFAULT_MAXIMUM_RETRIES` and `DEFAULT_DURATION_SLEEP_FOR_RETRY`) were at the beginning of the signature and not the end.

Reply:
Good point. Reordered arguments to put the "defaulted" ones last.